metadata/src/main/java/org/apache/kafka/metadata/publisher/ScramPublisher.java

```diff
@@ -55,10 +56,11 @@ public void onMetadataUpdate(MetadataDelta delta, MetadataImage newImage) {
             if (scramDelta != null) {
                 scramDelta.changes().forEach((mechanism, userChanges) -> {
                     userChanges.forEach((userName, change) -> {
+                        ScramMechanism internalMechanism = ScramMechanism.forMechanismName(mechanism.mechanismName());
```

Comment:
Why is this change needed?

Reply:
ScramMechanism.forMechanismName returns `org.apache.kafka.common.security.scram.internals.ScramMechanism` and CredentialProvider methods, i.e., removeCredential and updateCredential need `org.apache.kafka.common.security.scram.internals.ScramMechanism` type instead of `
org.apache.kafka.clients.admin.ScramMechanism`. 

`scramDelta.changes()` using `org.apache.kafka.clients.admin.ScramMechanism`.  That's why I have added that here.

Reply:
> I'd prefer if we keep the existing types. Let's focus this PR on the Scala to Java conversion.

Updated.